pkg/osquery/runtime/history/history.go

```diff
@@ -0,0 +1,106 @@
+package history
+
+import (
+	"errors"
+	"os"
+	"sync"
+	"time"
+)
+
+const maxInstances = 10
+
+var currentHistory *History = &History{}
+
+type History struct {
+	sync.Mutex
+	instances []*Instance
+}
+
+type NoInstancesError struct{}
+
+func (c NoInstancesError) Error() string {
+	return "no osquery instances have been added to history"
+}
+
+// GetHistory returns the last 10 instances of osquery started / restarted by launcher, each start / restart cycle is an entry
+func GetHistory() ([]Instance, error) {
+	if currentHistory.instances == nil {
+		return nil, NoInstancesError{}
+	}
+
+	currentHistory.Lock()
+	defer currentHistory.Unlock()
+
+	results := make([]Instance, len(currentHistory.instances))
+	for i, v := range currentHistory.instances {
+		results[i] = *v
+	}
+
+	return results, nil
+}
+
+// LatestInstance returns the latest osquery instance
+func LatestInstance() (Instance, error) {
```

Comment:
Why return a pointer from `lastInstance`? If you return the object, than `LastestInstance` can just be a wrapper.

Reply:
I get why `LastestInstance` returns an object. But why does `lastInstance` return a pointer? In fact, why have `lastInstance` at all?

Reply:
You're right, it's redundant, removed.